src/pseudopeople/utilities.py

```diff
@@ -58,22 +58,25 @@ def vectorized_choice(
 
 
 def get_index_to_noise(
-    data: Union[pd.DataFrame, pd.Series],
+    data: pd.DataFrame,
     noise_level: float,
     randomness_stream: RandomnessStream,
     additional_key: Any,
+    is_column_noise: bool = False,
 ) -> pd.Index:
     """
     Function that takes a series and returns a pd.Index that chosen by Vivarium Common Random Number to be noised.
     """
 
     # Get rows to noise
-    if isinstance(data, pd.Series):
-        not_empty_idx = data.index[(data != "") & (data.notna())]
+    if is_column_noise:
+        missing_idx = data.index[(data.isna().any(axis=1)) | (data.isin([""]).any(axis=1))]
```

Comment:
Looks good, but out of curiousity why did you switch to the opposite logic and then take the difference?

Reply:
Getting the missing_idx helped me get this working while debugging. I was having trouble getting and and all to work for the original logic. I also thought I may have to extract this line of code into a helper function for a mock like we did in one of the refactors but that wasn't the case.